src/datahike/db/transaction.cljc

```diff
@@ -295,15 +295,23 @@
               :entity entity
               :assertion acc}))))
 
-(defn- upsert-eid [db entity]
+(defn- upsert-eid [db entity tempids] ;; TODO: adjust to datascript?
```

Comment:
What's the difference to DataScript?

Reply:
It shows a clear separation of [resolution](https://github.com/tonsky/datascript/blob/78fdf8ac2666ff013a60460085c64d32f9573d00/src/datascript/db.cljc#L1224 ) and [validation](https://github.com/tonsky/datascript/blob/78fdf8ac2666ff013a60460085c64d32f9573d00/src/datascript/db.cljc#L1270) of upserts now: I'd also say that it is easier to read.